lib/ansible/modules/packaging/os/mas.py

```diff
@@ -0,0 +1,290 @@
+            msgs.append('Installed {0} app(s)'.format(self.count_install))
+        if self.count_upgrade > 0:
+            msgs.append('Upgraded {0} app(s)'.format(self.count_upgrade))
+        if self.count_uninstall > 0:
+            msgs.append('Uninstalled {0} app(s)'.format(self.count_uninstall))
+
+        if msgs:
+            self.result['changed'] = True
+            self.result['msg'] = ', '.join(msgs)
+
+        self.module.exit_json(**self.result)
+
+    def get_current_state(self, command):
+        ''' Returns the list of all app IDs; command can either be 'list' or 'outdated' '''
+
+        rc, raw_list, err = self.run([command])
+        rows = raw_list.split("\n")
+        list = {}
+        for r in rows:
+            # Format: "123456789 App Name"
+            r = r.split(' ', 1)
+            if len(r) == 2:
+                list[r[1]] = int(r[0])
+
+        return list
+
+    def installed(self):
+        ''' Returns the list of installed apps '''
+
+        # Populate cache if not already done
+        if self._installed is None:
+            self._installed = self.get_current_state('list')
+
+        return self._installed.values()
+
+    def is_installed(self, id):
+        ''' Checks whether the given app is installed '''
+
+        return int(id) in self.installed()
+
+    def is_outdated(self, id):
+        ''' Checks whether the given app is installed, but outdated '''
+
+        return int(id) in self.outdated()
+
+    def outdated(self):
+        ''' Returns the list of installed, but outdated apps '''
+
+        # Populate cache if not already done
+        if self._outdated is None:
+            self._outdated = self.get_current_state('outdated')
+
+        return self._outdated.values()
+
+    def run(self, cmd):
+        ''' Runs a command of the `mas` tool '''
+
+        cmd.insert(0, self.mas_path)
+        return self.module.run_command(cmd, False)
+
+    def upgrade_all(self):
+        ''' Upgrades all installed apps and sets the correct result data '''
+
+        outdated = self.outdated()
+
+        if not self.module.check_mode:
+            self.check_signin()
+
+            rc, out, err = self.run(['upgrade'])
+            if rc != 0:
+                self.module.fail_json(msg='Could not upgrade all apps: ' + out.rstrip())
+
+        self.count_upgrade += outdated
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            id=dict(type='list', elements='int'),
+            state=dict(type='str', default='present', choices=['absent', 'latest', 'present']),
+            upgrade_all=dict(type='bool', default=False, aliases=['upgrade']),
+        ),
+        supports_check_mode=True
+    )
+    mas = Mas(module)
+
+    if module.params['id']:
+        apps = module.params['id']
+    else:
+        apps = []
+
+    state = module.params['state']
+    upgrade = module.params['upgrade_all']
+
+    # Run operations on the given app IDs
+    for app in apps:
```

Comment:
```suggestion
    for app in sorted(set(apps)):
```
This way, you won't have trouble if apps show up more than once in the list (in which case `is_installed` and `is_outdated` can return wrong information), and the order is still deterministic also for older Python versions.